great_expectations/core/expectation_suite.py

```diff
@@ -145,19 +145,13 @@ def add_expectation(self, expectation: _TExpectation) -> _TExpectation:
             )
         should_save_expectation = self._has_been_saved()
         expectation_is_unique = all(
-            expectation.configuration != existing_expectation.configuration
-            for existing_expectation in self.expectations
+            expectation != existing_expectation for existing_expectation in self.expectations
         )
         if expectation_is_unique:
             # suite is a set-like collection, so don't add if it not unique
-            self.expectations.append(expectation)
             if should_save_expectation:
-                try:
-                    expectation = self._store.add_expectation(suite=self, expectation=expectation)
-                    self.expectations[-1].id = expectation.id
-                except Exception as exc:
-                    self.expectations.pop()
-                    raise exc  # noqa: TRY201
```

Comment:
We can just wait to append until we succeed with persistence - also, the `id` is already attached to the input expectation